sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -1647,33 +1611,60 @@ def get_transceiver_status(self):
-        trans_status['rxpowerlowalarm_flag'] = rx_power_flag_dict['rx_power_low_alarm']['RxPowerLowAlarmFlag1']
-        trans_status['rxpowerhighwarning_flag'] = rx_power_flag_dict['rx_power_high_warn']['RxPowerHighWarnFlag1']
-        trans_status['rxpowerlowwarning_flag'] = rx_power_flag_dict['rx_power_low_warn']['RxPowerLowWarnFlag1']
-        tx_bias_flag_dict = self.get_tx_bias_flag()
-        trans_status['txbiashighalarm_flag'] = tx_bias_flag_dict['tx_bias_high_alarm']['TxBiasHighAlarmFlag1']
-        trans_status['txbiaslowalarm_flag'] = tx_bias_flag_dict['tx_bias_low_alarm']['TxBiasLowAlarmFlag1']
-        trans_status['txbiashighwarning_flag'] = tx_bias_flag_dict['tx_bias_high_warn']['TxBiasHighWarnFlag1']
-        trans_status['txbiaslowwarning_flag'] = tx_bias_flag_dict['tx_bias_low_warn']['TxBiasLowWarnFlag1']
-        self.vdm_dict = self.get_vdm()
-        try:
-            trans_status['prefecberhighalarm_flag'] = self.vdm_dict['Pre-FEC BER Average Media Input'][1][5]
-            trans_status['prefecberlowalarm_flag'] = self.vdm_dict['Pre-FEC BER Average Media Input'][1][6]
-            trans_status['prefecberhighwarning_flag'] = self.vdm_dict['Pre-FEC BER Average Media Input'][1][7]
-            trans_status['prefecberlowwarning_flag'] = self.vdm_dict['Pre-FEC BER Average Media Input'][1][8]
-            trans_status['postfecberhighalarm_flag'] = self.vdm_dict['Errored Frames Average Media Input'][1][5]
-            trans_status['postfecberlowalarm_flag'] = self.vdm_dict['Errored Frames Average Media Input'][1][6]
-            trans_status['postfecberhighwarning_flag'] = self.vdm_dict['Errored Frames Average Media Input'][1][7]
-            trans_status['postfecberlowwarning_flag'] = self.vdm_dict['Errored Frames Average Media Input'][1][8]
-        except KeyError:
-            pass
+        if not self.is_flat_memory():
+            dp_state_dict = self.get_datapath_state()
+            for lane in range(1, self.NUM_CHANNELS+1):
+                trans_status['DP%dState' % lane] = dp_state_dict['DP%dState' % lane]
+            tx_output_status_dict = self.get_tx_output_status()
+            trans_status['txoutput_status'] = tx_output_status_dict['TxOutputStatus1']
+            rx_output_status_dict = self.get_rx_output_status()
+            for lane in range(1, self.NUM_CHANNELS+1):
+                trans_status['rxoutput_status_hostlane%d' % lane] = rx_output_status_dict['RxOutputStatus%d' % lane]
+            tx_fault = self.get_tx_fault()
+            trans_status['txfault'] = tx_fault[0]
+            tx_los = self.get_tx_los()
+            for lane in range(1, self.NUM_CHANNELS+1):
+                trans_status['txlos_hostlane%d' % lane] = tx_los[lane - 1]
+            tx_lol = self.get_tx_cdr_lol()
+            for lane in range(1, self.NUM_CHANNELS+1):
+                trans_status['txcdrlol_hostlane%d' % lane] = tx_lol[lane - 1]
+            rx_los = self.get_rx_los()
+            trans_status['rxlos'] = rx_los[0]
```

Comment:
cmis.py is applicable for non-ZR optics, so why only first lane?

Reply:
done